### ISSUE TITLE
Keep node headers in the include path even when node-addon-api is detected?

### ISSUE THREAD

**Issue by remyjette** [cmake-js/cmake-js#285]
Prior to #278, headers for `/include/node` (or `/src`, `/deps/v8/include`, `/deps/uv/include`) would be included even when nan/node-addon-api were located.

Now, this seems to be an either-or situation.

My project *mostly* uses node-addon-api, but there's one location where we need `uv.h`. Could there be a way for us to specify that we do need the node headers too?

Thanks!

**Comment by Julusian** [comment]
Yeah, I am open to adding this as an opt-in thing, with some warnings about not being truely node version safe in the readme.

I don't know this code well enough yet to know if that can be defined as a parameter, or if it needs to be read from package.json

**Comment by Maxime-J** [comment]
Same situation (Node-API + uv.h), hence the same issue.
Thanks for pointing in the right direction.

Here is a possible workaround:

First, remove binary property from package.json, advised to be added as
```json
"binary": {
  "napi_versions": [10]
}
```

That way, cmake-js downloads the appropriate headers+lib and sets the correct paths as CMAKE_JS variables.

But in that situation, cmake-js won't include Node-API headers directory, so we need to first retrieve the path and make sure it's included in `target_include_directories`:

```cmake
execute_process(
  COMMAND node -p "const path = require('node:path'); path.resolve(require('node-addon-api').include_dir);"
  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
  OUTPUT_VARIABLE CMAKE_JS_NAPI_INC
  OUTPUT_STRIP_TRAILING_WHITESPACE
)

target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_JS_NAPI_INC} ${CMAKE_JS_INC})
```

This is enough to make it work after following the README, `${CMAKE_JS_LIB}` being in `target_link_libraries`.

Note that the following block could then be safely removed as it would be ignored:
```cmake
if(MSVC AND CMAKE_JS_NODELIB_DEF AND CMAKE_JS_NODELIB_TARGET)
  # Generate node.lib
  execute_process(COMMAND ${CMAKE_AR} /def:${CMAKE_JS_NODELIB_DEF} /out:${CMAKE_JS_NODELIB_TARGET} ${CMAKE_STATIC_LINKER_FLAGS})
endif()
```
